src/chess/chess_game.rs

```diff
@@ -24,18 +26,88 @@ impl ChessMove {
     pub fn new(m: chess::ChessMove) -> Self {
         Self { m: m }
     }
+
     pub fn from_str(pos: &ChessPosition, move_text: &str) -> Result<Self, String> {
         return match chess::ChessMove::from_san(&pos.board, move_text) {
             Ok(m) => Ok(Self::new(m)),
             Err(e) => Err(err_to_str(e)),
         };
     }
+
+    pub fn get_raw(&self) -> &chess::ChessMove {
+        &self.m
+    }
+
+    pub fn to_idx(&self) -> usize {
+        let promotion = self.m.get_promotion();
+        if promotion.is_none() {
+            return self.m.get_source().to_index() * 64 + self.m.get_dest().to_index();
+        } else {
+            let promotions_base = 64 * 64;
+            let sf = self.m.get_source().get_file().to_index();
+            let df = self.m.get_dest().get_file().to_index();
+            let move_chunk = (sf * 2 + df) * 4;

```

Comment:
beseder

Reply:
this is just the inverse of:
"a7a8q","a7a8r","a7a8b","a7a8k","a7b8q","a7b8r","a7b8b","a7b8k",
"b7a8q","b7a8r","b7a8b","b7a8k","b7b8q","b7b8r","b7b8b","b7b8k",
"b7c8q","b7c8r","b7c8b","b7c8k","c7b8q","c7b8r","c7b8b","c7b8k",
"c7c8q","c7c8r","c7c8b","c7c8k","c7d8q","c7d8r","c7d8b","c7d8k",
"d7c8q","d7c8r","d7c8b","d7c8k","d7d8q","d7d8r","d7d8b","d7d8k",
"d7e8q","d7e8r","d7e8b","d7e8k","e7d8q","e7d8r","e7d8b","e7d8k",
"e7e8q","e7e8r","e7e8b","e7e8k","e7f8q","e7f8r","e7f8b","e7f8k",
"f7e8q","f7e8r","f7e8b","f7e8k","f7f8q","f7f8r","f7f8b","f7f8k",
"f7g8q","f7g8r","f7g8b","f7g8k","g7f8q","g7f8r","g7f8b","g7f8k",
"g7g8q","g7g8r","g7g8b","g7g8k","g7h8q","g7h8r","g7h8b","g7h8k",
"h7g8q","h7g8r","h7g8b","h7g8k","h7h8q","h7h8r","h7h8b","h7h8k"